### DISCUSSION TITLE
Saving User Meta with Signup

### DISCUSSION THREAD

**Discussion by yankeeinlondon** [supabase/supabase#560] [Questions]
I would like to have users signup with email/password and then encourage them strongly to confirm their email but at the same time I need to be able to capture more information about the user (in this case their first and last name) during the signup process. Ideally like to be able to choose between the two scenarios but right now I'm not sure if either are possible:

1. **Signup results in Login**. Upon signup, the user is also logged in but their status as _non-verified_ is noted. I see that both the `aud` and `role` properties are set to "authenticated" ... maybe one of these could be repurposed to indicate "needs-verfication" or something similar. Overall I think a property like `emailVerified` as a boolean flag would be the best solution but sometimes using existing semantics/fields helps. I think i read that you can _disable_ email verification and then you are logged in upon signup but I _do_ want to promote the need to signup and am happy to have the application take on the responsibility of limiting the logged in user until they have verified.
2. **Storing User Metadata**. There appears to be a `user_metadata` on the User table and I'm not sure how one is meant to access this or what guidelines are suggested. I know that "user profiles" are typically better placed into the public DB rather than being isolated to the authentication sandbox but if there were a way to place them there (even just temporarily), this could be of some use. 

I'm also open to other ways and means of solving this problem (which I would imagine is a pretty common use case) but these two solutions are just what I've seen in other solutions.

**Comment by silentworks** [comment]
I think the approach you could take is disabling the email verification in supabase and create your own `emailVerification` mechanism on your own `public.users` table. If the email address is wrong then you can always allow the user to update it by using `supabase.auth.update({ email: ""})`, or you could even add the `emailVerification` field to the `auth.users` metadata by using the same `supabase.auth.update({ data: { emailVerification: false })` as soon as the user has completed their signup. The only part I can't think of a solution for is actually sending the email for the user to verify their email, as this would require a different third party mailing system and a function to generate some sort of token.

I agree with you that email verification shouldn't stop a user from logging in, it should be left down to the app developer how they manage which features the user can use while not verified.

**Comment by J0** [comment]
We now support the notion of `email_confirmed` and  you should also be able to [save metadata on signup](https://supabase.com/docs/reference/javascript/auth-signup) by passing in a `data` option

Hope this helps and let me know if there are still issues!

**Comment by ioucyf** [comment]
Would you be so kind to elaborate more on the feature now supported or at least link to the corresponding page in the docs?

> We now support the notion of `email_confirmed` and  you should also be able to [save metadata on signup](https://supabase.com/docs/reference/javascript/auth-signup) by passing in a `data` option
> 
> 
> 
> Hope this helps and let me know if there are still issues!
